rosgraph_monitor/include/rosgraph_monitor/monitor.hpp

```diff
@@ -111,6 +111,10 @@ class RosGraphMonitor
 
   virtual ~RosGraphMonitor();
 
+  /// @brief Initialize the monitor and start watching the rosgraph with
+  /// configuration provided in constructor.
+  void init();
```

Comment:
We discussed - I'd rather pass the fully initialized config to the constructor than have a separated `init` function. RAII pattern makes for an easier API. So really I'd say the problem isn't with the `RosGraphMonitor` but instead with the `rosgraph_monitor::Node` init pattern